agent/engine/docker_task_engine_test.go

```diff
@@ -99,513 +119,123 @@ func mockSetupNSResult() *current.Result {
-	go func() { eventStream <- createDockerEvent(api.ContainerStopped) }()
-	go func() { eventStream <- createDockerEvent(api.ContainerStopped) }()
-
-	sleepTaskStop := testdata.LoadTask("sleep5")
-	sleepTaskStop.SetCredentialsID(credentialsID)
-	sleepTaskStop.SetDesiredStatus(api.TaskStopped)
-	taskEngine.AddTask(sleepTaskStop)
-	// As above, duplicate events should not be a problem
-	taskEngine.AddTask(sleepTaskStop)
-	taskEngine.AddTask(sleepTaskStop)
-
-	// Expect a bunch of steady state 'poll' describes when we trigger cleanup
-	client.EXPECT().RemoveContainer(gomock.Any(), gomock.Any()).Do(
-		func(removedContainerName string, timeout time.Duration) {
-			assert.Equal(t, createdContainerName, removedContainerName, "Container name mismatch")
-		}).Return(nil)
-
-	imageManager.EXPECT().RemoveContainerReferenceFromImageState(gomock.Any())
-	metadataManager.EXPECT().Clean(gomock.Any()).Return(nil)
-	// trigger cleanup
-	cleanup <- time.Now()
-	go func() { eventStream <- createDockerEvent(api.ContainerStopped) }()
-
-	// Wait for the task to actually be dead; if we just fallthrough immediately,
-	// the remove might not have happened (expectation failure)
-	for {
-		tasks, _ := taskEngine.(*DockerTaskEngine).ListTasks()
-		if len(tasks) == 0 {
-			break
-		}
-		time.Sleep(5 * time.Millisecond)
-	}
-}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			metadataConfig := defaultConfig
+			metadataConfig.TaskCPUMemLimit = config.ExplicitlyDisabled
+			metadataConfig.ContainerMetadataEnabled = true
+			ctrl, client, mockTime, taskEngine, credentialsManager, imageManager, metadataManager := mocks(t, &metadataConfig)
+			defer ctrl.Finish()
 
-// TestCotnainerMetadataEnabledErrorPath checks case when metadata service is enabled but calls return errors
-func TestContainerMetadataEnabledErrorPath(t *testing.T) {
-	metadataConfig := defaultConfig
-	metadataConfig.ContainerMetadataEnabled = true
-	ctrl, client, mockTime, taskEngine, credentialsManager, imageManager, metadataManager := mocks(t, &metadataConfig)
-	defer ctrl.Finish()
+			roleCredentials := credentials.TaskIAMRoleCredentials{
+				IAMRoleCredentials: credentials.IAMRoleCredentials{CredentialsID: "credsid"},
+			}
+			credentialsManager.EXPECT().GetTaskCredentials(credentialsID).Return(roleCredentials, true).AnyTimes()
+			credentialsManager.EXPECT().RemoveCredentials(credentialsID)
+
+			sleepTask := testdata.LoadTask("sleep5")
+			sleepTask.SetCredentialsID(credentialsID)
+
+			eventStream := make(chan DockerContainerChangeEvent)
+			// createStartEventsReported is used to force the test to wait until the container created and started
+			// events are processed
+			createStartEventsReported := sync.WaitGroup{}
+
+			client.EXPECT().Version().Return("1.12.6", nil)
+			client.EXPECT().ContainerEvents(gomock.Any()).Return(eventStream, nil)
+			containerName := make(chan string)
+			go func() {
```

Comment:
Can this be just `go setCreatedContainerName(<-containerName)`?

Reply:
yeah, sure